### DISCUSSION TITLE
Can I use php@8.0 fpm by fastcgi_pass?

### DISCUSSION THREAD

**Discussion by eyedroot** [shivammathur/homebrew-php#87] [Q&A / Help] [answered]
Thank you. I am using it well on Mac via shivammathur/php@8.0.

I have one question. Is there any way to connect with nginx installed through brew install nginx and shivammathur/php@8.0?

```nginx
 65         location ~ \.php$ {
 66             root           html;
 67             fastcgi_pass   127.0.0.1:9000;
 68             fastcgi_index  index.php;
 69             fastcgi_param  SCRIPT_FILENAME  /scripts$fastcgi_script_name;
 70             include        fastcgi_params;
 71         }
```

Thanks again.

**Comment by shivammathur** [comment, marked as answer]
Here is the process I follow to setup Nginx after PHP setup.

- Check php-fpm's port, by default it should be `9000`.
```
sudo lsof -i -n -P | grep php-fpm
```
- Install `Nginx` and test it.
```
brew install nginx
sudo nginx
curl http://localhost:8080
```
- Edit `Nginx` config `/usr/local/etc/nginx/nginx.conf`
  - I would change port 8080 to 80.
     ```diff
     - listen 8080;
     + listen 80;
     ```
  - Update index files.
     ```diff
     - index index.html;
     + index index.html index.htm index.php;
     ```
  - Add directives to connect `Nginx` with `php-fpm`. Use the port from step 1 if you have a different port than `9000`.
     ```conf
     location ~ \.php$ {
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_split_path_info ^(.+\.php)(/.+)$;
        fastcgi_pass 127.0.0.1:9000;
        fastcgi_index index.php;
        include fastcgi_params;
     }
     ```
- Reload `Nginx` and test it.
```
sudo nginx -s reload
sudo lsof -i -n -P | grep nginx
curl http://localhost
```
- Remove `index.html` and Create a `index.php` file with `phpinfo()` to test `Nginx` and `php-fpm` config.
```
rm /usr/local/var/www/index.html
echo "<?php echo phpinfo(); ?>" | tee /usr/local/var/www/index.php
```
- Go to `http://localhost` in your browser and you should get php configuration.

**Reply by eyedroot** [reply]
Thank you very much. I knew I had to install php-fpm separately, but it seems that installing php@8.0 will work together!
